### PR TITLE
Rewards contract set up

### PR DESCRIPTION
**Pull request by shriyatyagii**
Added rewards contract with the following functions - `claimMorpho`, `claimCompound`, `claimAave`.

### CODE REVIEW

File: contracts/mainnet/connectors/morpho-rewards/main.sol [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	 * @param _proof The merkle proof that validates this claim.
+	 */
+	function claimMorpho(address _account, uint256 _claimable, bytes32[] calldata _proof)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		require(_proof.length > 0, "proofs-empty");
+
+		MORPHO_REWARDS.claim(_account, _claimable, _proof);
```
Comment (Rubilmax) [inline contracts/mainnet/connectors/morpho-rewards/main.sol:32]: Shouldn't the amount of claimed MORPHO tokens be stored somewhere in the storage via `setUint`?

**Review by QGarchery — COMMENTED** [review]
LGTM ! :tada:

File: contracts/mainnet/connectors/morpho-rewards/main.sol (hunk 1 above)
Comment (MerlinEgalite) [reply contracts/mainnet/connectors/morpho-rewards/main.sol:32]: Agree I think it's missing too

File: contracts/mainnet/connectors/morpho-rewards/main.sol [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			_tradeForMorphoToken,
+			_amountOfRewards,
+			_setId
+		);
+	}
+}
+
+contract ConnectV2MorphoCompound is MorphoRewards {
+	string public constant name = "Morpho-Rewards-v1.0";
+}
```
Comment (MerlinEgalite) [inline contracts/mainnet/connectors/morpho-rewards/main.sol]: The reason it's not done with Aave is because there's no rewards, right?

Comment (shriyatyagii) [reply contracts/mainnet/connectors/morpho-rewards/main.sol]: Hey! Yes, that was the motive but now that you've mentioned it, I believe it is not providing a decent description. So I've renamed it to a generic `ConnectV2MorphoRewards`

File: contracts/mainnet/connectors/morpho-rewards/main.sol (hunk 1 above)
Comment (shriyatyagii) [reply contracts/mainnet/connectors/morpho-rewards/main.sol:32]: Yes! You're right. I've added it. My bad, I missed it.

**Review by MerlinEgalite — COMMENTED** [review]
LGTM now!

**Review by Rubilmax — COMMENTED** [review]
LTGM!

**Comment by shriyatyagii** [comment]
Deployed - [0x19a369cd6d9b0ec809dca84309138c7f1e7563f7](https://etherscan.io/address/0x19a369cd6d9b0ec809dca84309138c7f1e7563f7#code)
Name - `MORPHO-REWARDS-A`

Tx - [0xb31557ee872ff1c6f5cc1986a09cb3b1e2af3a0896f7d96b43d9492e0f05c477](https://etherscan.io/tx/0xb31557ee872ff1c6f5cc1986a09cb3b1e2af3a0896f7d96b43d9492e0f05c477)

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**